modules/imgcodecs/src/grfmt_jpeg.cpp

```diff
@@ -510,7 +511,7 @@ bool  JpegDecoder::readData( Mat& img )
                 for( int iy = 0 ; iy < m_height; iy ++ )
                 {
                     uchar* data = img.ptr<uchar>(iy);
-                    jpeg_read_scanlines( cinfo, buffer, 1 );
+                    num_read_lines += jpeg_read_scanlines( cinfo, buffer, 1 );
```

Comment:
I diged a bit in documentation and usage samples. Looks like the original patch was correct. We set maxLines=1 and should get only one scanline at a time. In case if `num_read_lines` is used, you have to change for loop condition and Size for collor conversions to accommodate. Also it breaks non-contiguous Mats.